libqtile/core/manager.py

```diff
@@ -530,19 +539,52 @@ def current_window(self):
     def add_strut(self, strut):
         from libqtile.bar import Bar, Gap
 
+        try:
+            if strut[0]:
+                x = strut[0]
+            elif strut[1]:
+                x = self.x_screen_dimensions.get('x') - strut[1]
+            else:
+                x = ((strut[8] + strut[9]) / 2 or (strut[10] + strut[11]) / 2)
+            if strut[2]:
+                y = strut[2]
+            elif strut[3]:
+                y = self.x_screen_dimensions.get('y') - strut[2]
+            else:
+                y = ((strut[4] + strut[5]) / 2 or (strut[6] + strut[7]) / 2)
+            screen = self.find_screen(x, y)
+        except IndexError:  # if _NET_WM_STRUT_PARTIAL is not set, we need to default to something
```

Comment:
Is the idea that we are just falling back to the short version upon any indexing failure into `strut`? If so, it might be logically neater if the `try` block is replaced with a test of `len(strut)`, and may be easier to debug if needed in the future